godtools/App/Features/AppLanguage/Presentation/LanguageSettings/Subviews/ToolLanguagesAvailableOffline/ToolLanguagesAvailableOfflineView.swift

```diff
@@ -40,41 +40,40 @@ struct ToolLanguagesAvailableOfflineView: View {
             SeparatorView()
                 .padding([.top], 11)
             
-            ScrollView(.vertical, showsIndicators: true) {
+            GeometryReader { scrollViewGeometry in
                 
-                VStack(alignment: .leading, spacing: 0) {
-                    
-                    ToolLanguageAvailableOfflineLanguageView()
-                    
-                    ToolLanguageAvailableOfflineLanguageView()
-                    
-                    ToolLanguageAvailableOfflineLanguageView()
-                    
-                    ToolLanguageAvailableOfflineLanguageView()
-                    
-                    ToolLanguageAvailableOfflineLanguageView()
-                    
-                    ToolLanguageAvailableOfflineLanguageView()
-                    
-                    ToolLanguageAvailableOfflineLanguageView()
+                let buttonHeight: CGFloat = 50
+                let spaceBelowScrollView: CGFloat = 25
+                let scrollViewMaxHeight: CGFloat = scrollViewGeometry.size.height - buttonHeight - spaceBelowScrollView
```

Comment:
The problem before was that the scrollView took up the maximum space and always forced the blue button to the bottom of the screen, even if there were only a few rows in the scrollView:
<img src="https://github.com/CruGlobal/godtools-swift/assets/8647064/838dbd72-5bf1-486a-83d4-a47cde367e09" width="200">
Whereas Figma has the button directly beneath the last row:
<img width="200" alt="Screenshot 2024-02-02 at 3 56 18 PM" src="https://github.com/CruGlobal/godtools-swift/assets/8647064/d8147d1d-2b87-42af-8d75-7e51c54040b9">


I was trying to figure out a way to do this without calculations, but just doing `.fixedSize(horizontal: false, vertical: true)` without also setting the maxHeight ends up pushing the blue button off the screen if there's too many rows.  I can play with this some more and see if there's a better solution

Reply:
Hey @rachaelblue I see what you're saying now about the button sticking to the bottom of the screen originally.

It would be nice to have the scrollview wrap it's content, but then you're right, it seems that would introduce the possibility of the button getting pushed off screen.

It would be nice if there were some solution where that height didn't have to be calculated by explicitly telling it which spacings and element heights needed removing.  It does open the possibility of creating a UI bug in the future.  If you find yourself spending too much time here then I'd say create a ticket for it and it can be re-investigate later.

Reply:
Tried a few different things here, but none quite had the desired result.  It seems that SwiftUI doesn't let you make the ScrollView shrink to fit its content while also ensuring the view stays within the safeArea, unless you explicitly set a maxHeight to keep it within the safe area.  Went ahead and made a ticket: https://jira.cru.org/browse/GT-2287

Reply:
Hey @rachaelblue thanks for checking into that some more.  You're right we would need some way to tell it not to expand beyond the safe area.  Thank you for making the ticket too, maybe something will be available in the future to allow for that behavior.